src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Visitors/ExpressionTreeRewriter.cs

```diff
@@ -440,10 +440,10 @@ private Expr GenerateBuiltInBinaryOperator(ExprBinOp expr)
                 typeL = convertL;
                 didEnumConversion = true;
             }
-            else if (typeL.IsNullableType() && typeL.StripNubs().isEnumType())
+            else if (typeL is NullableType nubL && nubL.UnderlyingType.isEnumType())
             {
                 Debug.Assert(expr.IsLifted);
-                convertL = GetSymbolLoader().GetTypeManager().GetNullable(typeL.StripNubs().underlyingEnumType());
```

Comment:
`StripNubs` and its `out bool` variant is a safe form in that it returns `this` when called on a type other than a nullable, so it's reasonable enough except perhaps in regards to its name.